BERT/main.py

```diff
@@ -157,12 +197,27 @@ def run_main(args):
     #     WorkerModule("worker2", pipeline_on_devices(1, 3, 5, 7, n_encoders=args.nlayers // 2, include_head=True), MLMTaskSharder, ntokens, args.emsize, args.nhead, args.nhid, args.dropout),
     # )
 
-    model = DistributedCPURPCSequential(
-        WorkerModule("worker1", pipeline_on_devices(5, 2, include_embeddings=True, n_encoders=args.nlayers // 6, n_encoders_on_last_gpu=args.nlayers // 6), MLMTaskSharder, ntokens, args.emsize, args.nhead, args.nhid, args.dropout),
-        WorkerModule("worker2", pipeline_on_devices(0, 7, n_encoders=args.nlayers // 3), MLMTaskSharder, ntokens, args.emsize, args.nhead, args.nhid, args.dropout),
-        WorkerModule("worker3", pipeline_on_devices(1, 4, n_encoders=args.nlayers // 3), MLMTaskSharder, ntokens, args.emsize, args.nhead, args.nhid, args.dropout),
-        WorkerModule("worker4", pipeline_on_devices(6, 3, n_encoders=args.nlayers // 6, n_encoders_on_first_gpu=args.nlayers // 6, include_head=True), MLMTaskSharder, ntokens, args.emsize, args.nhead, args.nhid, args.dropout),
-    )
+    layers = [RemoteModule("w0/cuda:0", MLMTaskEmbedding, (ntokens, args.emsize))]
+    n_encoders = args.nlayers
+    if not False:
+      for i, device in enumerate([f"w{i}/cuda:{i}" for i in (0, 1, 2, 3, 4, 5, 6)]):
+        this_encoders = n_encoders // (7-i)
+        layers.append(RemoteModule(device, MLMTaskEncoder, (args.emsize,  args.nhead, args.nhid, this_encoders, args.dropout)))
+        n_encoders -= this_encoders
+    layers.append(RemoteModule("w7/cuda:7", MLMTaskHead, (ntokens, args.emsize)))
+    #layers.append(RemoteModule("w3/cuda:3", NoOp, ()))
+    org_model = nn.Sequential(*layers)
+
+    #org_model = nn.Sequential(*(
+    #        MLMTaskSharder(["w5/cuda:5", "w2/cuda:2"], dict(include_embeddings=True, n_encoders=args.nlayers // 6, n_encoders_on_last_gpu=args.nlayers // 6), ntokens, args.emsize, args.nhead, args.nhid, args.dropout)
+    #        +MLMTaskSharder(["w0/cuda:0", "w7/cuda:7"], dict(n_encoders=args.nlayers // 3), ntokens, args.emsize, args.nhead, args.nhid, args.dropout)
+    #        +MLMTaskSharder(["w1/cuda:1", "w4/cuda:4"], dict(n_encoders=args.nlayers // 3), ntokens, args.emsize, args.nhead, args.nhid, args.dropout)
+    #        +MLMTaskSharder(["w6/cuda:6", "w3/cuda:3"], dict(n_encoders=args.nlayers // 6, n_encoders_on_first_gpu=args.nlayers // 6, include_head=True), ntokens, args.emsize, args.nhead, args.nhid, args.dropout)
+    #    ))
+
+    graph = make_graph(org_model)
+    #for node in graph.nodes: print(node.module.on, node.get_name())
+    model = DistributedPipeline(graph, chunks=8)
```

Comment:
instead of creating `graph` and `model` this way, just do: `model=Pipe(org_model)`
for old implementation